### ISSUE TITLE
Is s3 secret enabed?

### ISSUE THREAD

**Issue by hanqiushi** [seaweedfs/seaweedfs-operator#70]
Can I enable accessKey/secretKey for s3 now ?

**Comment by scrayos** [comment]
Yes, you can. But (as far as I know), you have to add it manually by running something like:
```
s3.configure -access_key=any -secret_key=any -buckets=bucket1 -user=me -actions=Read,Write,List,Tagging,Admin -apply
```
in the `weed shell` of one of the master nodes.

Source: https://github.com/seaweedfs/seaweedfs/wiki/Amazon-S3-API#s3-authentication

**Comment by scrayos** [comment]
> Thanks, I've done this in one of master nodes, but I can still access filer nodes without using accessKey/secretKey, did I miss something ?

I am by no means knowledgeable regarding filer, but my guess would be that the accessKey/secretKey is only mandatory for the specific bucket and not the filer as a whole. 

If you want to secure your whole filer instance, my guess would be, that you had to configure `jwt.filer_signing.key` and `jwt.filer_signing.read.key` in `security.toml` ([Reference](https://github.com/seaweedfs/seaweedfs/wiki/Security-Configuration)).

But as I said, I'm only guessing here. But even if I was right, a new property needs to be added to the Seaweed Custom Resource Definition, so that users would be able to set security.toml options. But input from someone more knowleadgable would be highly appreciated.

**Comment by chrislusf** [comment]
This is supported now. You can configure S3 access key / secret key authentication via the standalone S3 gateway's `configSecret`, which references a Kubernetes Secret containing the S3 identities config.

Example:

```yaml
apiVersion: seaweed.seaweedfs.com/v1
kind: Seaweed
metadata:
  name: seaweed1
spec:
  s3:
    replicas: 1
    configSecret:
      name: my-s3-config
      key: seaweedfs_s3_config.json
```

Create the Secret from a config file like [`config/samples/seaweedfs_s3_config.json`](https://github.com/seaweedfs/seaweedfs-operator/blob/master/config/samples/seaweedfs_s3_config.json):

```json
{
  "identities": [
    {
      "name": "user1",
      "actions": ["Admin"],
      "credentials": [
        { "accessKey": "testkey", "secretKey": "testpass" }
      ]
    }
  ]
}
```

```
kubectl create secret generic my-s3-config --from-file=seaweedfs_s3_config.json
```

The operator mounts the Secret and passes it to `weed s3` via `-config`. The same mechanism is also available on the (deprecated) filer-embedded S3 via `spec.filer.s3.configSecret`.

For inter-component security (the `security.toml` follow-up question), the operator now auto-generates `security.toml` with JWT signing keys whenever a filer/admin is configured or TLS is enabled, so that no longer needs manual setup either.

Closing as resolved.
